modules/programs/swayr.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.swayr;
+  tomlFormat = pkgs.formats.toml { };
+  configFile = tomlFormat.generate "config.toml" cfg.settings;
+  finalConfig = pkgs.writeText "swayr.toml"
+    ((builtins.readFile configFile) + cfg.extraConfig);
+in {
+  meta.maintainers = [ lib.hm.maintainers."9p4" ];
+
+  options.programs.swayr = {
+    enable = mkEnableOption "the swayr service";
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
```

Comment:
yoiu haven't address my comment about the default: this generates an empty config by default which breaks swayrd, please set better defaults.

Reply:
ha I see you've done `mkIf (cfg.settings != { }) { source = finalConfig; };` . I wonder if I can use `xdg.configFile."swayr/config.toml".source = mkForce null` or sthg along those lines so that you dont have to modify the module.

Reply:
nevermind that I can just do `xdg.configFile."swaync/config.json".enable = false;` and bypass the thing. Apologies for the noise.